tools/validator.py

```diff
@@ -80,14 +80,18 @@ def CheckFile(self, filename):
         for source in artifact_definition.sources:
           if source.type_indicator in (
               definitions.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
-            if self._HasDuplicateRegistryKeyPaths(
-                filename, artifact_definition, source):
+
+            # Exempt the legacy file from duplicate checking because it has
+            # duplicates intentionally.
+            if (filename != os.path.join('definitions', 'legacy.yaml') and
```

Comment:
I opt not to embed function calls in other logic too much makes debugging hard
legacy_yaml_filename = os.path.join('definitions', 'legacy.yaml')
maybe even make this a class constant?

if (filename != legacy_yaml_filename and self._HasDuplicateRegistryKeyPaths(
   ...